telemetry_core/src/main/java/com/newrelic/telemetry/SenderConfiguration.java

```diff
@@ -67,36 +69,77 @@ public static SenderConfigurationBuilder builder(String defaultUrl, String baseP
     private boolean auditLoggingEnabled = false;
     private String secondaryUserAgent;
 
+    /** Create a new Builder, with the default URL and base path. */
     public SenderConfigurationBuilder(String defaultUrl, String basePath) {
       this.defaultUrl = defaultUrl;
       this.basePath = basePath;
     }
 
+    /**
+     * Configure the New Relic Insert API key to use.
+     *
+     * @return this builder;
+     */
     public SenderConfigurationBuilder apiKey(String apiKey) {
       this.apiKey = apiKey;
       return this;
     }
 
+    /**
+     * Configure the {@link HttpPoster} implementation to use.
+     *
+     * @return this builder
+     */
     public SenderConfigurationBuilder httpPoster(HttpPoster httpPoster) {
       this.httpPoster = httpPoster;
       return this;
     }
 
+    /**
+     * Configure the *full* endpoint URL for data to be sent to, including the path.
+     *
+     * @param sendUrl A full {@link URL}, including the path.
+     * @return this builder.
+     */
     public SenderConfigurationBuilder endpointUrl(URL sendUrl) {
       this.endpointUrl = sendUrl;
       return this;
     }
 
+    /**
+     * Configure the endpoint URL for data to be sent to, not including the path. If the path is
+     * provided, it will be replaced with the default path.
+     *
+     * @param endpointUri A {@link URI}, path will be ignored.
+     * @return this builder.
+     */
+    public SenderConfigurationBuilder endpoint(URI endpointUri) throws MalformedURLException {
```

Comment:
If this option is only scheme + host + port, then I would recommend that it accept three parameters. Otherwise, the choice between `endpoint` and `endpointUrl` is completely unclear. Without carefully reading the Javadoc (which should not be required), the fact that one accepts path and one does not is unexpected.

Alternative solution is to name them `endpoint(URL)` and `endpointIgnoringPath(URL)`.

Reply:
please discuss in #156